stat.go

```diff
@@ -125,6 +125,35 @@ func ChiSquare(obs, exp []float64) float64 {
 	return result
 }
 
+// CircularMean returns the circular mean of the dataset.
+//	atan2(\sum_i w_i * sin(alpha_i), \sum_i w_i * cos(alpha_i))
+// If weights is nil then all of the weights are 1. If weights is not nil, then
+// len(x) must equal len(weights).
+func CircularMean(x, weights []float64) float64 {
+	if weights != nil && len(x) != len(weights) {
+		panic("stat: slice length mismatch")
+	}
+
+	var aX, aY float64
+	if weights != nil {
+		for i := range x {
+			aX += weights[i] * math.Cos(x[i])
```

Comment:
Updated. Is there a difference in performance? I've noticed other places in `stat.go` follow the one-element range syntax (which is why I followed suit).